Program_Demo.R

```diff
@@ -0,0 +1,166 @@
+# student is not found, the function prints an error message and returns the
+# unchanged list.
+edit_student_attendance <- function(id, new_status) {
+  for (i in 1:length(student_list)) {
+    student <- student_list[[i]]
+    if (student[1] == id) {
+      student[3] <- new_status
+      student_list[[i]] <- student
+      cat("Attendance updated successfully.")
+      return(student_list)
+    }
+  }
+}
+
+
+# Function to add a student to the student_list list data structure.
+# The function accepts an id number and student name and creates a vector
+# using the provided student details. This vector is then added to the
+# student_list list.
+add_student <- function(id, student_name) {
+  student <- c(id, student_name, "Present")
+  student_list[[length(student_list) + 1]] <- student
+  cat("Student added successfully.")
+  return(student_list)
+}
+
+# Function to display student attendance with its corresponding
+# ID number, Name, and remark. This includes the current date
+# of the current day. The function takes the data of the
+# 'student_list' vector and then converts it to a data frame
+# and then finally formats and displays it in a tabular format
+display_attend <- function() {
+  #Data of Added Students
+  student_data <- data.frame(
+    ID_Num = sapply(student_list, "[[", 1),
+    Name = sapply(student_list, "[[", 2),
+    Status = sapply(student_list, "[[", 3)
+  )
+
+  # Get the number of rows in data of student
+  num_students <- nrow(student_data)
+
+  # Concatenate the Class Attendance and Current Date Function
+  heading <- paste("<-------- CLASS ATTENDANCE-", Sys.Date(),"-------->\n")
+
+  # Display heading
+  cat(heading)
+
+  # Define the headers
+  header <- c("ID_Num", "Name", "Status")
+
+  # Create and Display the Formatted Header
+  formatted_header <- sprintf("%-10s\t%-20s\t%-1s\n", header[1], header[2], header[3])
+  cat(formatted_header)
+  cat("\n")
+
+  # Display the Formatted Student's Attendance Data with for loop
+  for (i in 1:num_students) {
+    formatted_rows <- sprintf("%-10s\t%-20s\t%-1s\n",
+                              student_data$ID_Num[i],
+                              student_data$Name[i],
+                              student_data$Status[i])
+    cat(formatted_rows)
+  }
+}
+
+# While loop where options for the program are displayed and functions
+# of the program are invoked depending on the user's chosen option.
+while (TRUE) {
+
+  cat("Attendance Monitoring System\n")
+  cat("1. Add a Student\n")
+  cat("2. Edit Student Attendance\n") 
+  cat("3. Display Student Attendance\n")
+  cat("4. Exit\n")
+
+  choice <- as.integer(readline("Choose the option that corresponds to your choice: "))
+
+  # Add a Student Option
+  if (choice == 1) {
+
+    #Validate Entered ID Number
+    while(TRUE){
+      id <- readline("Enter student ID: ")
+
+
+      if (is.na(as.integer(id))) {
+        cat("ID must be numeric!")
+      } else {
+        id <- as.integer(id)  # Convert id to an integer
```

Comment:
Please refer to Section III of the company's coding standards under ORGANIZATION - Commenting Guidelines regarding simple comments and ensure compliance with it. Make sure to use the updated commenting guidelines format consistently across the entire code to meet the modified standard. Thank you.